RELEASE.md

```diff
@@ -0,0 +1,92 @@
+## Crafting a new release
+
+### Merge in upstream release to openshift repo:
+
+```bash
+# Check that a remote reference to openshift and upstream exists
+$ git remote -v | grep -e 'openshift\|upstream'
+openshift	git@github.com:openshift/knative-client.git (fetch)
+openshift	git@github.com:openshift/knative-client.git (push)
+upstream	https://github.com/knative/client.git (fetch)
+upstream	https://github.com/knative/client.git (push)
+
+# Create an openshift branch pointing to openshift client's master
+# (if not already done)
+$ git co -b openshift openshift/master
+
+# Go to scripts dir
+$ cd openshift/release
+
+# Create new release branch. Parameters are the upstream release tag
+# and the name of the branch to create
+# Usage: ./create-release-branch.sh <upstream-tag> <downstream-release-branch>
+# <upstream-tag>: The tag referring the upstream release
+# <downstream-release-branch>: Name of the release branch to create
+$ ./create-release-branch.sh v0.9.0 release-v0.9.0
+
+# Push release branch to openshift/knative-client repo
+$ git push openshift release-v0.9.0
+```
+
+### Create a ci-operator configuration and Prow job configurations
+
+* Create a fork of https://github.com/openshift/release (if not already done)
+* Create a new config YML for your release. Copy over `release-next.yaml` and adapt it by updating the image name from `knative-nightly` to a version specific name `knative-v0.9.0`
+* Once the CI passes for the the release branch and is ready to go for QA, push the tag for the release for in the format `openshift-`<version>, e.g. `openshift-v0.9.0`
+*
+```bash
+# Jump into the knative client config directory in the openshift/release
+$ cd ci-operator/config/openshift/knative-client
+
+# Copy over the nightly builds config to a release specific config with
+# the name of the yaml file ends with the new release branch name (e.g. release-v0.9.0)
+$ cp openshift-knative-client-release-next.yaml openshift-knative-client-release-v0.9.0.yaml
+
+# Adapt the configuration for a new image name
+# - Change .promotion.name to a release specific name (knative-v0.9.0)
+$ vi openshift-knative-client-release-v0.9.0.yaml
+
+# Jump to top-level repo directory
+$ cd ../../../../
+
+# Call Prow job generator via Docker. You need a local Docker daemon installed
+$ docker run -it -v $(pwd)/ci-operator:/ci-operator:z  \
+     registry.svc.ci.openshift.org/ci/ci-operator-prowgen:latest \
+     --from-dir /ci-operator/config --to-dir /ci-operator/jobs
+
+# Add the image mirroring settings
+$ vi core-services/image-mirroring/knative/mapping_knative_v0_9_quay
+
+# Add a line for the kn image like
+# registry.svc.ci.openshift.org/openshift/knative-v0.9.0:knative-client quay.io/openshift-knative/knative-client:v0.9.0
```

Comment:
@navidshaikh Is it like this or more

```
registry.svc.ci.openshift.org/openshift/knative-v0.9.0:kn quay.io/openshift-knative/kn:v0.9.0
```

?

Reply:
(asking, because on quay.io, there is a `openshift-knative/kn` but no `openshift-knative/knative-client`)